File: Client/Tuckshop.WebApp/src/Domain/Models/Customers/Commands/DepositFunds.ts
import { Attributes, ModelBase, Rules, Validation } from '@singularsystems/neo-core';

export default class DepositFunds extends ModelBase {
    static typeName = "DepositFunds";

    constructor() {
        super();
        this.makeObservable();
    }

    @Rules.Required()
    public customerId: number = 0;

    @Attributes.Float()
    @Rules.Required()
    public amount: number = 0;

    // Client only properties / methods

    protected static addBusinessRules(rules: Validation.Rules<DepositFunds>) {
        super.addBusinessRules(rules);
    }

    public toString(): string {
        if (this.isNew) {
            return "New deposit funds";
        } else {
            return "Deposit Funds";
        }
    }
}

File: Server/Tuckshop.Domain/Tuckshop.Core.Models/Models/Customers/Commands/DepositFunds.cs
namespace Tuckshop.Core.Models.Customers.Commands
{
  using Neo.Model;
  using Neo.Model.Validation;
  using System.ComponentModel.DataAnnotations;
  using System.ComponentModel.DataAnnotations.Schema;

  /// <summary>
  /// Command to deposit funds into a customer's wallet
  /// </summary>
  public class DepositFunds : ModelBase<DepositFunds>
  {
    /// <summary>
    /// Gets or sets the Customer Id.
    /// </summary>
    [Required]
    public int CustomerId { get; set; }

    /// <summary>
    /// Gets or sets the amount to deposit
    /// </summary>
    [Required]
    [Column(TypeName = "money")]
    public decimal Amount { get; set; }

    /// <inheritdoc/>
    protected override void AddBusinessRules(ValidationRules<DepositFunds> rules)
    {
      base.AddBusinessRules(rules);
      rules.FailWhen(c => c.Amount <= 0, "Amount must be greater than zero");
    }
  }
}
